README.md

```diff
@@ -1,2 +1,227 @@
+}
+ */
+let optimizedDocument = optimizer.getOptimizedDocument({rules: {reuseComponents: true,removeComponents: true,moveToComponents: true }})
+/*
+the optimizedDocument value will be:
+
+asyncapi: 2.0.0
+info:
+  title: Streetlights API
+  version: '1.0.0'
+
+channels:
+
+  smartylighting/event/{streetlightId}/lighting/measured:
+    parameters:
+      streetlightId:
+        $ref: "#/components/schemas/streetlightId"
+    subscribe:
+      operationId: receiveLightMeasurement
+      traits:
+        - bindings:
+            kafka:
+              clientId: my-app-id
+      message:
+        name: lightMeasured
+        title: Light measured
+        contentType: application/json
+        traits:
+          - headers:
+              type: object
+              properties:
+                my-app-header:
+                  type: integer
+                  minimum: 0
+                  maximum: 100
+        payload:
+          type: object
+          properties:
+            lumens:
+              type: integer
+              minimum: 0
+            #full form is used, we can ref it to: #/components/schemas/sentAt
+            sentAt:
+              $ref: "#/components/schemas/sentAt"
+
+  smartylighting/action/{streetlightId}/turn/on:
+    parameters:
+      streetlightId:
+        $ref: "#/components/schemas/streetlightId"
+    publish:
+      operationId: turnOn
+      traits:
+        - bindings:
+            kafka:
+              clientId: my-app-id
+      message:
+        name: turnOnOff
+        title: Turn on/off
+        traits:
+          - headers:
+              type: object
+              properties:
+                my-app-header:
+                  type: integer
+                  minimum: 0
+                  maximum: 100
+        payload:
+          type: object
+          properties:
+            sentAt:
+              $ref: "#/components/schemas/sentAt"
+
+components:
+  parameters:
+    streetlightId:
+      schema:
+      type: string
+  schemas:
+    #this schema is ref-ed in one channel and used full form in another. library should be able to identify and ref the second channel as well.
+    sentAt:
+      type: string
+      format: date-time`
+ */
+```
+
+## API
+
+### Constructor
+
+#### new Optimizer(document)
+
+`document` is a mandatory object that is parsed with `@asyncapi/parser`:
+
+### Methods
+
+#### getReport() : OptimizerReport
+#### getOptimizedDocument([options]) : string
+`options` is an OPTIONAL object that contains the following customizations:
+* `rules`: using rules different optimization types can be enabled or disabled.
+    * `reuseComponents` if set to true, optimizer will apply all of *reuseComponents* changes from report. (default: *false*)
+    * `removeComponents` if set to true, optimizer will apply all of *removeComponents* changes from report. (default: *false*)
+    * `moveToComponents` if set to true, optimizer will apply all of *moveToComponents* changes from report. (default: *false*) 
```

Comment:
In the https://github.com/asyncapi/optimizer/issues/4 issue you have set true as default value for `reuseComponents` and `moveToComponents`. Why you change it? I think that everything (by default) should be set to true.

Reply:
Sorry about that, I meant to make them true.😆